### PR TITLE
DENG-815 Add backfill info cli command

### PR DESCRIPTION
**Pull request by wwyc**
[DENG 815](https://mozilla-hub.atlassian.net/browse/DENG-815):  

* Added new backfill info cli command to get information on backfills with specific status.  This command can be used to retrieve a list of backfills with Drafting status that needs to be processed.
* Added tests for backfill info command
* Added utils to retrieve backfills given qualified_table_name

### CODE REVIEW

File: bigquery_etl/cli/utils.py [hunk 1]
```diff
@@ -120,6 +128,56 @@ def paths_matching_name_pattern(pattern, sql_path, project_id, files=["*.sql"]):
...
+            "Qualified table name must be named like:" + " <project>.<dataset>.<table>"
+        )
+
+    return (project_id, dataset_id, table_id)
+
+
+def backfills_matching_name_pattern(
+    sql_dir, project_id, qualified_table_name
+) -> defaultdict[Path, List[Backfill]]:
+    """Return backfills with name pattern."""
```
Comment (ANich) [inline bigquery_etl/cli/utils.py:148]: nit: could you say what the keys of the returned dict are supposed to represent here (or in the function name?) I assume it's the path to the backfill file? Overall this function seems more like a backfill util than a CLI util

**Review by ANich — APPROVED** [review]
Left a few comments but overall LGTM - I'd include in the PR description why this command is useful

File: bigquery_etl/cli/backfill.py [hunk 2]
```diff
@@ -220,3 +226,75 @@ def validate(
...
+                excluded_dates = {entry.excluded_dates}
+                """.rstrip()
+
+            backfill_str += f"""
+                reason = {entry.reason}
+                watcher(s) = {entry.watchers}
+                status = {entry.status.value}
+                """
+
+            click.echo(backfill_str)
```
[the remark this answers is not in the document]
Comment (wwyc) [reply bigquery_etl/cli/backfill.py]: Sounds good!  I've added a `__str__` method.

File: bigquery_etl/cli/utils.py (hunk 1 above)
Comment (wwyc) [reply bigquery_etl/cli/utils.py:148]: Yes the keys represent the path to the backfill file.

**Review by ANich — APPROVED** [review]

File: bigquery_etl/cli/backfill.py [hunk 3]
```diff
@@ -220,3 +222,64 @@ def validate(
...
+    """,
+)
+@click.argument("qualified_table_name", required=False)
+@sql_dir_option
+@project_id_option("moz-fx-data-shared-prod")
+@click.option(
+    "--status",
+    type=click.Choice([s.value.lower() for s in BackfillStatus]),
+    help="Filter backfills with this status.",
+    default=None,
```
[the remark this answers is not in the document]
Comment (wwyc) [reply bigquery_etl/cli/backfill.py:251]: Right.  I removed the default.

**Pull request merged**